crypto/bn/gcd.c

```diff
@@ -458,12 +465,15 @@ static BIGNUM *BN_mod_inverse_no_branch(BIGNUM *out, int *out_no_inverse,
   A->neg = 0;
 
   if (B->neg || (BN_ucmp(B, A) >= 0)) {
-    /* Turn BN_FLG_CONSTTIME flag on, so that when BN_div is invoked,
+    /* Ensure BN_FLG_CONSTTIME flag is on, so that when BN_div is invoked,
      * BN_div_no_branch will be called eventually.
      */
-    pB = &local_B;
-    BN_with_flags(pB, B, BN_FLG_CONSTTIME);
-    if (!BN_nnmod(B, pB, A, ctx)) {
+    assert(BN_get_flags(B, BN_FLG_CONSTTIME) != 0);
+    if (BN_get_flags(B, BN_FLG_CONSTTIME) == 0) {
+      OPENSSL_PUT_ERROR(BN, 100000);  // FIXME: what error code?
+      goto err;
+    }
```

Comment:
I suggest just moving the `BN_set_flags` call for `B` here and avoiding this pattern. The above pattern should only be used for `BIGNUM`s that were passed in as parameters to a function or structure members.